diff -uNr pam_mount-0.9.20.orig/src/mount.c pam_mount-0.9.20/src/mount.c
--- pam_mount-0.9.20.orig/src/mount.c	2004-07-23 03:57:21.000000000 +0200
+++ pam_mount-0.9.20/src/mount.c	2004-11-24 11:13:42.538189016 +0100
@@ -121,7 +121,7 @@
 /* FIXME: this function may need to be broken up and made more readable */
 #ifdef HAVE_LIBCRYPTO
 	int ret = 1;
-	size_t segment_len;
+	int segment_len;
 	unsigned char ct_fs_key[MAX_PAR];	/* encrypted filesystem key. */
 	size_t ct_fs_key_len;
 	unsigned char hashed_authtok[EVP_MAX_KEY_LENGTH];	/* hash(system authtok) */
